gen_initramfs.sh

```diff
@@ -1808,34 +1837,38 @@ append_modules() {
 		gen_die "${error_message}"
 	fi
 
-	determine_KEXT
-
 	cd "${modules_srcdir}" || gen_die "Failed to chdir to '${modules_srcdir}'!"
 
-	print_info 2 "$(get_indent 2)modules: >> Copying modules from '${modules_srcdir}' to initramfs ..."
+	print_info 2 "$(get_indent 2)modules: >> Searching modules in '${modules_srcdir}' ..."
 
-	local i= mymod=
-	local n_copied_modules=0
-	for i in $(gen_dep_list)
+	local mymod=
+	local -a modlist=()
+	while IFS= read -r -u 3 mymod
 	do
-		mymod=$(find . -name "${i}${KEXT}" 2>/dev/null | head -n 1)
-		if [ -z "${mymod}" ]
+		if [ -z "${mymod}" -o "${mymod}" = '(builtin)' ]
 		then
-			print_warning 3 "$(get_indent 3) - ${i}${KEXT} not found; Skipping ..."
-			continue;
+			continue
 		fi
 
-		print_info 3 "$(get_indent 3) - Copying ${i}${KEXT} ..."
-		cp -ax --parents --target-directory "${modules_dstdir}" "${mymod}" 2>/dev/null \
-			|| gen_die "Failed to copy '${modules_srcdir}/${mymod}' to '${modules_dstdir}'!"
-		n_copied_modules=$[$n_copied_modules+1]
-	done
+		if [ ! -f "${mymod}" ]
+		then
+			print_warning 3 "$(get_indent 3) - ${i}${KEXT} is missing; Ignoring ..."
+			continue
+		fi
+
+		modlist+=( "${mymod/#${modules_srcdir}\//}" )
+	done 3< <( (
+		modinfo -b "${KERNEL_MODULES_PREFIX%/}" -k "${KV}" -F filename ${GK_INITRAMFS_KMODULES} 2>/dev/null
```

Comment:
does this work in the cross-compile kernels?

Reply:
I tried to copy `/lib/modules/*-gentoo-dist` dir into `/mnt/prefix`, and `modinfo -b ... -k ...` worked.

Reply:
I plan to make improvements to this place in the future. Something like:
`modinfo ... || echo` - handle bad status from `modinfo` in the `while`.
But for this I will need to make some changes to the `gen_moddeps.sh`. This will increase the predictability of the script. I think it would be better to make it a separate commit.
Currently, everything should work reliably, unexpected errors should be logged and filtered in `determine_initramfs_kmodules()`.